topic-configuration/README.adoc

```diff
@@ -0,0 +1,689 @@
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`1 GB`
+
+h|Kafka property name
+|`segment.bytes`
+|===
+--
+
+Segment time::
++
+--
+Segment time is the amount of time after which the current log segment is rolled even if the segment file is not full. This property ensures that the topic cleanup policy can delete or compact old messages as needed.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`604800000 ms` (7 days)
+
+h|Kafka property name
+|`segment.ms`
+|===
+--
+
+Segment jitter time::
++
+--
+Segment jitter time is the maximum delay for log segment rolling. This delay prevents bursts of log segment rolling activity.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`0 seconds`
+
+h|Kafka property name
+|`segment.jitter.ms`
+|===
+--
+
+File delete delay::
++
+--
+File delete delay is the amount of time that a file is retained in the system before the file is deleted.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`60000 ms` (1 minute)
+
+h|Kafka property name
+|`file.delete.delay.ms`
+|===
+--
+
+Preallocate log segment files::
++
+--
+Preallocate log segment files determines whether to preallocate the file on disk when creating a new log segment. This property ensures sufficient disk space for log segments.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Boolean
+
+h|Default value
+|`Disabled`
+
+h|Kafka property name
+|`preallocate`
+|===
+--
+
+=== Index
+
+These properties control the indexing of the log.
+
+Index interval size::
++
+--
+Index interval size is the number of bytes between each index entry to its offset index. The default setting indexes a message about every 4096 bytes. More indexing enables reads to be closer to the exact position in the log but makes the index larger.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`4 KB`
+
+h|Kafka property name
+|`index.interval.bytes`
+|===
+--
+
+Segment index size::
++
+--
+Segment index size is the size of the index that maps offset to file positions.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`10 MB`
+
+h|Kafka property name
+|`segment.index.bytes`
+|===
+--
+
+=== Flush
+
+These properties control the frequency of the flushing of the log.
+
+Flush interval messages::
++
+--
+Flush interval messages is the number of messages between each data flush to the log.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`9223372036854775807 ms` (292271023 years, 2 weeks, 2 days, 13 hours, 12 minutes, 56.064 seconds)
+
+h|Kafka property name
+|`flush.messages`
+|===
+--
+
+Flush interval time::
++
+--
+Flush interval time is the amount of time between each data flush to the log.
```

Comment:
Same as above. It's the interval between flushing of the log cache to disk?

Reply:
Ditto